runtime/binding-http-filesystem/src/main/java/io/aklivity/zilla/runtime/binding/http/filesystem/internal/config/HttpFileSystemWithResolver.java

```diff
@@ -68,20 +73,36 @@ public HttpFileSystemWithResult resolve(
         }
         String16FW path = new String16FW(path0);
 
-        HttpFileSystemWithResult result = null;
-
         HttpHeaderFW method = httpBeginEx.headers().matchFirst(h -> HEADER_METHOD_NAME.equals(h.name()));
+        int capabilities = 0;
         if (method != null)
         {
             if (HEADER_METHOD_VALUE_HEAD.equals(method.value()))
             {
-                result = new HttpFileSystemWithResult(path, HEADER_METHOD_MASK_HEAD);
+                capabilities = HEADER_METHOD_MASK_HEAD;
             }
             else if (HEADER_METHOD_VALUE_GET.equals(method.value()))
             {
-                result = new HttpFileSystemWithResult(path, HEADER_METHOD_MASK_GET);
+                capabilities = HEADER_METHOD_MASK_GET;
+            }
+        }
+        HttpHeaderFW ifNotMatched = httpBeginEx.headers().matchFirst(h -> HEADER_IF_NONE_MATCH_NAME.equals(h.name()));
+        String16FW etag = new String16FW("");
+        if (ifNotMatched != null)
+        {
+            String16FW value = ifNotMatched.value();
+            etag = etagRO.wrap(value.buffer(), value.offset(), value.limit());
+        }
+        HttpHeaderFW prefer = httpBeginEx.headers().matchFirst(h -> HEADER_PREFER_NAME.equals(h.name()));
+        int timeout = 0;
+        if (prefer != null)
+        {
+            Matcher timeoutMatcher = timoutMatcher.reset(prefer.value().asString());
+            if (timeoutMatcher.find())
+            {
+                timeout = Integer.parseInt(timeoutMatcher.group(1));
             }
         }
-        return result;
+        return new HttpFileSystemWithResult(path, capabilities, etag, timeout);
```

Comment:
Pass last argument as `TimeUnit.SECONDS.toMillis(wait)` so we have the timeout in millis for use with the signaler.